Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp

```diff
@@ -685,20 +685,24 @@ UiProfile Spire::make_text_box_profile() {
       auto& read_only = get<bool>("read_only", profile.get_properties());
       read_only.connect_changed_signal([text_box] (auto is_read_only) {
         text_box->set_read_only(is_read_only);
+        text_box->adjustSize();
       });
       auto& current = get<QString>("current", profile.get_properties());
       current.connect_changed_signal([=] (const auto& current) {
         if(text_box->get_model()->get_current() != current) {
           text_box->get_model()->set_current(current);
+          text_box->adjustSize();
         }
       });
       auto& placeholder = get<QString>("placeholder", profile.get_properties());
       placeholder.connect_changed_signal([text_box] (const auto& text) {
         text_box->set_placeholder(text);
       });
-      text_box->get_model()->connect_current_signal([&] (const auto& value) {
-        current.set(value);
-      });
+      text_box->get_model()->connect_current_signal(
+        [&, text_box = text_box] (const auto& value) {
```

Comment:
Why is text_box = text_box used here?

Reply:
'current' is captured by reference as the default capture , but if I capture text_box by reference it's referencing a pointer that is going out of scope. So in this case I copy the text_box pointer into another pointer that's also called text_box that is used within the lambda.

Reply:
I think [&, text_box] or [=, &current] is better.